src/storage/btree.c

```diff
@@ -18725,6 +18727,46 @@ btree_range_opt_check_add_index_key (THREAD_ENTRY * thread_p, BTREE_SCAN * bts,
 	}
     }
 
+  /* resolve domains */
+  if (multi_range_opt->sort_col_dom == NULL)
+    {
+      multi_range_opt->sort_col_dom =
+	(TP_DOMAIN **) db_private_alloc (thread_p, multi_range_opt->num_attrs * sizeof (TP_DOMAIN *));
+      if (multi_range_opt->sort_col_dom == NULL)
+	{
+	  error = ER_OUT_OF_VIRTUAL_MEMORY;
+	  goto exit;
+	}
+
+      for (i = 0; i < multi_range_opt->num_attrs; i++)
+	{
+	  multi_range_opt->sort_col_dom[i] = &tp_Null_domain;
+	}
+      multi_range_opt->has_null_domain = true;
+    }
+
+  if (multi_range_opt->has_null_domain)
+    {
+      has_null_domain = false;
+      for (i = 0; i < multi_range_opt->num_attrs; i++)
+	{
+	  assert (multi_range_opt->sort_col_dom[i] != NULL);
+	  if (multi_range_opt->sort_col_dom[i] == &tp_Null_domain)
+	    {
+	      domain = tp_domain_resolve_value (&new_key_value[i], NULL);
+	      if (domain != &tp_Null_domain)
+		{
+		  multi_range_opt->sort_col_dom[i] = domain;
+		}
+	      else
+		{
+		  has_null_domain = true;
```

Comment:
If we have null domain it means that all values are null. We have to change the null domain once we have a not null value. Otherwise, we can't compare not null values.